### DISCUSSION TITLE
Libraries created / updated to support RASPBERRY_PI_PICO_W using CYW43439 WiFi

### DISCUSSION THREAD

**Discussion by khoih-prog** [earlephilhower/arduino-pico#757] [General]
Just released [**WiFiWebServer** v1.9.0](https://github.com/khoih-prog/WiFiWebServer/releases/tag/v1.9.0) to support **RASPBERRY_PI_PICO_W** using CYW43439 WiFi.

To be used with the new arduino-pico core version **v2.4.0+**

Please try and report issue, bug, etc.

Thanks,

---

### Releases v1.9.0

1. Add support to RASPBERRY_PI_PICO_W using CYW43439 WiFi
2. Update `Packages' Patches`

**Comment by khoih-prog** [comment]
From Arduino Forum's [Libraries created / updated to support new RASPBERRY_PI_PICO_W boards](https://forum.arduino.cc/t/libraries-created-updated-to-support-new-raspberry-pi-pico-w-boards/1021990)

These libraries are created or updated to add support to new [**RASPBERRY_PI_PICO_W**](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html#raspberry-pi-pico-w) boards, using [**Earle Philhower's arduino-pico core v2.4.0+**](https://github.com/earlephilhower/arduino-pico). 

![image](https://user-images.githubusercontent.com/57012152/184524162-178b98ea-509b-40ae-b50a-c41c0c145992.png)

The lists will be updated gradually.

---

### Modified Libraries

- [x]  1.  [**WiFiMulti_Generic**](https://github.com/khoih-prog/WiFiMulti_Generic). From v1.2.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/WiFiMulti_Generic.svg)](https://github.com/khoih-prog/WiFiMulti_Generic/releases/latest)
- [x]  2.  [**WiFiWebServer**](https://github.com/khoih-prog/WiFiWebServer). From v1.9.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/WiFiWebServer.svg)](https://github.com/khoih-prog/WiFiWebServer/releases/latest)
- [x]  3.  [**AsyncMQTT_Generic**](https://github.com/khoih-prog/AsyncMQTT_Generic). From v1.6.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/AsyncMQTT_Generic.svg)](https://github.com/khoih-prog/AsyncMQTT_Generic/releases/latest)
- [x]  4.  [**RPI_PICO_TimerInterrupt**](https://github.com/khoih-prog/RPI_PICO_TimerInterrupt). From v1.1.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/RPI_PICO_TimerInterrupt.svg)](https://github.com/khoih-prog/RPI_PICO_TimerInterrupt/releases/latest).
- [x]  5.  [**TimerInterrupt_Generic**](https://github.com/khoih-prog/TimerInterrupt_Generic). From v1.6.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/TimerInterrupt_Generic.svg)](https://github.com/khoih-prog/TimerInterrupt_Generic/releases/latest)
- [x]  6.  [**WebSockets2_Generic**](https://github.com/khoih-prog/WebSockets2_Generic). From v1.13.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/WebSockets2_Generic.svg)](https://github.com/khoih-prog/WebSockets2_Generic/releases/latest)
- [x]  7.  [**WebSockets_Generic**](https://github.com/khoih-prog/WebSockets_Generic). From v2.16.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/WebSockets_Generic.svg)](https://github.com/khoih-prog/WebSockets_Generic/releases/latest)
- [x]  8.  [**MQTTPubSubClient_Generic**](https://github.com/khoih-prog/MQTTPubSubClient_Generic). From v1.1.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/MQTTPubSubClient_Generic.svg)](https://github.com/khoih-prog/MQTTPubSubClient_Generic/releases/latest)

### New Libraries

- [x]  1.  [**AsyncTCP_RP2040W**](https://github.com/khoih-prog/AsyncTCP_RP2040W). From v1.0.0,  [![GitHub release](https://img.shields.io/github/release/khoih-prog/AsyncTCP_RP2040W.svg)](https://github.com/khoih-prog/AsyncTCP_RP2040W/releases/latest)
- [x]  2.  [**AsyncWebServer_RP2040W**](https://github.com/khoih-prog/AsyncWebServer_RP2040W). From v1.0.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/AsyncWebServer_RP2040W.svg)](https://github.com/khoih-prog/AsyncWebServer_RP2040W/releases/latest)
- [x]  3.  [**AsyncHTTPRequest_RP2040W**](https://github.com/khoih-prog/AsyncHTTPRequest_RP2040W). From v1.0.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/AsyncHTTPRequest_RP2040W.svg)](https://github.com/khoih-prog/AsyncHTTPRequest_RP2040W/releases/latest)
- [x]  4.  [**AsyncUDP_RP2040W**](https://github.com/khoih-prog/AsyncUDP_RP2040W). From v1.0.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/AsyncUDP_RP2040W.svg)](https://github.com/khoih-prog/AsyncUDP_RP2040W/releases/latest)
- [x]  5.  [**AsyncDNSServer_RP2040W**](https://github.com/khoih-prog/AsyncDNSServer_RP2040W). From v1.0.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/AsyncDNSServer_RP2040W.svg)](https://github.com/khoih-prog/AsyncDNSServer_RP2040W/releases/latest)
- [x]  6.  [**WiFiManager_RP2040W_Lite**](https://github.com/khoih-prog/WiFiManager_RP2040W_Lite). From v1.6.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/WiFiManager_RP2040W_Lite.svg)](https://github.com/khoih-prog/WiFiManager_RP2040W_Lite/releases/latest)
- [x]  7.  [**WiFiManager_RP2040W**](https://github.com/khoih-prog/WiFiManager_RP2040W). From v1.0.0. [![GitHub release](https://img.shields.io/github/release/khoih-prog/WiFiManager_RP2040W.svg)](https://github.com/khoih-prog/WiFiManager_RP2040W/releases/latest)

**Reply by FeelMedia** [reply]
Great stuff - can we expect to see WifiManager ported too? ;-)

**Reply by khoih-prog** [reply]
Yes, in the bucket list.

Something similar to [WiFiManager_NINA_Lite](https://github.com/khoih-prog/WiFiManager_NINA_Lite) or [WiFiManager_RTL8720](https://github.com/khoih-prog/WiFiManager_RTL8720).

If possible, AsyncTCP_SSL_RP2040W, then AsyncHTTPSRequest_RP2040W.

**Comment by khoih-prog** [comment]
Just released [WiFiManager_RP2040W_Lite](https://github.com/khoih-prog/WiFiManager_RP2040W_Lite)

---

### Initial Release v1.6.0

1. Add support to RP2040W boards, using [`arduino-pico core`](https://github.com/earlephilhower/arduino-pico)
2. Bump version to v1.6.0 to sync with [WiFiManager_Portenta_H7_Lite library](https://github.com/khoih-prog/WiFiManager_Portenta_H7_Lite)

**Comment by khoih-prog** [comment]
Just released full-fledged [**WiFiManager_RP2040W** library](https://github.com/khoih-prog/WiFiManager_RP2040W)

---

### Initial Release v1.0.0

1. Add support to RP2040W boards, using [`arduino-pico core`](https://github.com/earlephilhower/arduino-pico). To be used with new core having **DNSServer** library (**later than v2.4.0**)

**Comment by khoih-prog** [comment]
Just released [**WebSockets2_Generic** v1.13.0](https://github.com/khoih-prog/WebSockets2_Generic/releases/tag/v1.13.0) to support **RP2040W**

---

### Release v1.13.0

1. Add support to `RP2040W` using `CYW43439 WiFi` or `Ethernet` with `arduino-pico` core

**Comment by khoih-prog** [comment]
Just released [**WebSockets_Generic** v2.16.0](https://github.com/khoih-prog/WebSockets_Generic/releases/tag/v2.16.0) to support **RP2040W** using `ws` or `BearSSL wss`

---

### Release v2.16.0

1. Add support to **RP2040-based boards, such as Nano_RP2040_Connect, RASPBERRY_PI_PICO, ADAFRUIT_FEATHER_RP2040**, using WiFiNINA or Ethernet
2. Add `WS` and `BearSSL WSS` support to `RP2040W` using `CYW43439 WiFi` with `arduino-pico` core
3. Optional user-defined 
- `WEBSOCKETS_TCP_TIMEOUT`, default 5,000ms
- `EIO_HEARTBEAT_INTERVAL`, default 20,000ms
- `SIO_PING_INTERVAL`, default 60,000ms
- `SIO_PONG_TIMEOUT`, default 90,000ms
- `SIO_DISCONNECT_TIMEOUT_COUNT`, default 5
4. Update `Packages' Patches`

**Comment by khoih-prog** [comment]
Just released [**MQTTPubSubClient_Generic** v1.1.0](https://github.com/khoih-prog/MQTTPubSubClient_Generic/releases/tag/v1.1.0) to support **RP2040W**

---

### Releases v1.1.0

1. Add support to `RP2040W` using `MQTT`, `Secured MQTT` or `MQTT-over-WebSockets` with [**arduino-pico** core](https://github.com/earlephilhower/arduino-pico)
2. Update `Packages' Patches`

**Comment by khoih-prog** [comment]
### [AsyncWebServer_RP2040W](https://github.com/khoih-prog/AsyncWebServer_RP2040W)

#### Release v1.4.0

1. Add `LittleFS` functions such as AsyncFSWebServer
2. Add examples [AsyncFSWebServer](https://github.com/khoih-prog/AsyncWebServer_RP2040W/tree/main/examples/AsyncFSWebServer) and [AsyncFSWebServer_Complex](https://github.com/khoih-prog/AsyncWebServer_RP2040W/tree/main/examples/AsyncFSWebServer_Complex) to demo the new feature

---

#### AsyncFSWebServer_Complex on RASPBERRY_PI_PICO_W using CYW43439 WiFi

Following is debug terminal output when running example [**AsyncFSWebServer_Complex**](examples/AsyncFSWebServer_Complex) on RASPBERRY_PI_PICO_W using CYW43439 WiFi.


```
Start AsyncFSWebServer_Complex on RASPBERRY_PI_PICO_W with RP2040W CYW43439 WiFi
AsyncTCP_RP2040W v1.1.0
AsyncWebServer_RP2040W v1.4.0
Connecting to SSID: HueNet1
SSID: HueNet1
Local IP Address: 192.168.2.77
Opening / directory
FS File: CanadaFlag_1.png, size: 40.25KB
FS File: CanadaFlag_2.png, size: 8.12KB
FS File: CanadaFlag_3.jpg, size: 10.89KB
FS File: css, size: 0B
FS File: edit.htm.gz, size: 4.02KB
FS File: favicon.ico, size: 1.12KB
FS File: graphs.js.gz, size: 1.92KB
FS File: index.htm, size: 3.63KB
FS File: js, size: 0B

AsyncWebServer started @192.168.2.77
Open http://192.168.2.77/edit to see the file browser
AsyncFSEditor::handleRequest: Sending AsyncWebServerResponse

```


You can access the Async_WebSockets Servers at the displayed server IP, e.g. `192.168.2.77`

<p align="center">
    <img src="https://github.com/khoih-prog/AsyncWebServer_RP2040W/raw/main/pics/AsyncFSWebServer_Complex.png">
</p>

**Comment by ZinnerC** [comment]
@khoih-prog Hi. Great work. Is there any chance that these libraries could be used via LWIP and a W5100 for a wired connection?
